File: HanYeojin/Assignment2/yeojin/cafe-server/src/main/java/com/yeojin/cafe/repository/OrderRepository.java
package com.yeojin.cafe.repository;

import com.yeojin.cafe.domain.Order;
import org.springframework.stereotype.Repository;

import java.util.*;
import java.util.concurrent.atomic.AtomicLong;

@Repository
public class OrderRepository {
    private final Map<Long, Order> orderStore = new HashMap<>();
    private final AtomicLong sequence = new AtomicLong(1);

    // 주문 저장
    public Order saveOrder(Order order) {
        order.setId(sequence.getAndIncrement());
        orderStore.put(order.getId(), order);
        return order;
    }

    // order ID로 조회
    public Optional<Order> findOrderById(Long id) {
        return Optional.ofNullable(orderStore.get(id));
    }
}

File: assignment2/MinSeungho/cafe/src/main/kotlin/com/seungho/cafe/repository/OrderRepository.kt
package com.seungho.cafe.repository

import com.seungho.cafe.domain.Order
import org.springframework.stereotype.Repository
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.atomic.AtomicLong

@Repository
class OrderRepository {
    private val store = ConcurrentHashMap<Long, Order>()
    private val sequence = AtomicLong(0L)

    fun save(order: Order): Order {
        val newId = sequence.incrementAndGet()
        val newOrder = order.copy(id = newId)
        store[newId] = newOrder
        return newOrder
    }

    fun findById(id: Long): Order? {
        return store[id]
    }

    fun findAll(): List<Order> {
        return store.values.toList()
    }
}